airflow/providers/amazon/aws/hooks/minio.py

```diff
@@ -0,0 +1,80 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Interact with MinIO Simple Storage Service (S3)."""
+from __future__ import annotations
+
+from functools import cached_property
+from typing import Any
+
+import boto3
+
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+
+
+class MinIOHook(S3Hook):
+    """Interact with MinIO Simple Storage Service (S3)."""
+
+    conn_name_attr = "minio_conn_id"
+    default_conn_name = "minio_default"
+    conn_type = "minio"
+    hook_name = "MinIO storage service"
+
+    def __init__(
+        self,
+        minio_conn_id: str = default_conn_name,
+        transfer_config_args: dict | None = None,
+        extra_args: dict | None = None,
+        *args,
+        **kwargs,
+    ) -> None:
+        kwargs["transfer_config_args"] = transfer_config_args
```

Comment:
Why doing so? If you remove this property, the behaviour is the exact same one to me